asgi_rabbitmq/core.py

```diff
@@ -921,31 +826,57 @@ def make_fernet(self, key):
 
     async def send(self, channel, message):
         """Send the message to the channel."""
+        # Typecheck
+        assert isinstance(message, dict), 'message is not a dict'
         assert self.valid_channel_name(channel), 'Channel name is not valid'
-        return await wrap_future(self.thread.schedule(SEND, channel, message))
+        # Make sure the message does not contain reserved keys
+        assert '__asgi_channel__' not in message
+        # If it's a process-local channel, strip off local part and stick full
+        # name in message.
+        channel_non_local_name = channel
+        if '!' in channel:
```

Comment:
is this required? rabbitmq adapter handles process local channels differently than the redis adapter

Reply:
Cool, it seems it's not needed. I was tricked by default implementation of `valid_channel_name(receive=True)` [forbidding local channels on receive](https://github.com/django/channels/blob/a1ecd5ee72a538f19bdd9e8f6bb91bb8aabba5d9/channels/layers.py#L153-L156), while afais `receive()` is in fact called with local channels.